src/solo12.cpp

```diff
@@ -314,7 +314,7 @@ void Solo12::wait_until_ready()
 
 bool Solo12::is_ready()
 {
-    return state_ != Solo12State::ready;
+    return state_ == Solo12State::ready;
```

Comment:
Check if there isn't a wait until ready that uses this...

Reply:
There is a `wait_until_ready()` method in the same class, but it doesn't use `is_ready()` (and at least on first glance, its implementation looks correct):
https://github.com/open-dynamic-robot-initiative/solo/blob/89d1dc0f227c74563b818611a74471da51c08475/src/solo12.cpp#L299-L313

Apart from this, `is_ready()` only seems to be used in `solo12_hardware_calibration.cpp`.  I can check if this still works, next time I'm in the lab.

Reply:
great thanks!!!
otherwise of course we should merge this PR

Reply:
I tested `solo12_hardware_calibration` now. From what I can judge, it still works, or better works correctly now (without the fix it prints the "robot is ready" message too early).  So I think this can be merged.

Reply:
You can merge at your convenience